Compatibility with dns 2.8.0rc1

I've run unittests for localzone with dnspython 2.8.0rc1 and that uncovered the following issue:

```
============================= test session starts ==============================
platform linux -- Python 3.14.0rc2, pytest-8.3.5, pluggy-1.6.0
cachedir: .tox/py314/.pytest_cache
rootdir: /builddir/build/BUILD/python-localzone-0.9.8-build/localzone-0.9.8
collected 22 items

tests/test_context.py ...                                                [ 13%]
tests/test_models.py .........F.....                                     [ 81%]
tests/test_util.py ....                                                  [100%]

=================================== FAILURES ===================================
_______________________ test_zone_add_record_no_content ________________________

    def test_zone_add_record_no_content():
        with localzone.manage(ZONEFILE, ORIGIN) as z:
            with pytest.raises((AttributeError, DNSSyntaxError)):
>               z.add_record("test", "txt", None)

tests/test_models.py:82: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
localzone/models.py:195: in add_record
    rdata = dns.rdata.from_text(rdclass, rdtype, content, origin=self.origin)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

rdclass = <RdataClass.IN: 1>, rdtype = <RdataType.TXT: 16>, tok = None
origin = <DNS name example.com.>, relativize = True, relativize_to = None
idna_codec = None

    def from_text(
        rdclass: dns.rdataclass.RdataClass | str,
        rdtype: dns.rdatatype.RdataType | str,
        tok: dns.tokenizer.Tokenizer | str,
        origin: dns.name.Name | None = None,
        relativize: bool = True,
        relativize_to: dns.name.Name | None = None,
        idna_codec: dns.name.IDNACodec | None = None,
    ) -> Rdata:
        """Build an rdata object from text format.
    
        This function attempts to dynamically load a class which
        implements the specified rdata class and type.  If there is no
        class-and-type-specific implementation, the GenericRdata class
        is used.
    
        Once a class is chosen, its from_text() class method is called
        with the parameters to this function.
    
        If *tok* is a ``str``, then a tokenizer is created and the string
        is used as its input.
    
        *rdclass*, a ``dns.rdataclass.RdataClass`` or ``str``, the rdataclass.
    
        *rdtype*, a ``dns.rdatatype.RdataType`` or ``str``, the rdatatype.
    
        *tok*, a ``dns.tokenizer.Tokenizer`` or a ``str``.
    
        *origin*, a ``dns.name.Name`` (or ``None``), the
        origin to use for relative names.
    
        *relativize*, a ``bool``.  If true, name will be relativized.
    
        *relativize_to*, a ``dns.name.Name`` (or ``None``), the origin to use
        when relativizing names.  If not set, the *origin* value will be used.
    
        *idna_codec*, a ``dns.name.IDNACodec``, specifies the IDNA
        encoder/decoder to use if a tokenizer needs to be created.  If
        ``None``, the default IDNA 2003 encoder/decoder is used.  If a
        tokenizer is not created, then the codec associated with the tokenizer
        is the one that is used.
    
        Returns an instance of the chosen Rdata subclass.
    
        """
        if isinstance(tok, str):
            tok = dns.tokenizer.Tokenizer(tok, idna_codec=idna_codec)
        if not isinstance(tok, dns.tokenizer.Tokenizer):
>           raise ValueError("tok must be a string or a Tokenizer")
E           ValueError: tok must be a string or a Tokenizer

/usr/lib/python3.14/site-packages/dns/rdata.py:772: ValueError
=========================== short test summary info ============================
FAILED tests/test_models.py::test_zone_add_record_no_content - ValueError: to...
========================= 1 failed, 21 passed in 0.12s =========================
```

The failing check has been added in: https://github.com/rthalley/dnspython/commit/fee25b8ec2dcb323c33a6257099a23804a95a215

--- a/tests/test_models.py
+++ b/tests/test_models.py
@@ -78,7 +78,7 @@ def test_zone_add_record_unknown_type():
 
 def test_zone_add_record_no_content():
     with localzone.manage(ZONEFILE, ORIGIN) as z:
-        with pytest.raises((AttributeError, DNSSyntaxError)):
+        with pytest.raises((AttributeError, DNSSyntaxError, ValueError)):
             z.add_record("test", "txt", None)
 
 

